solrmonitor/solrmonitor.go

```diff
@@ -140,11 +150,25 @@ func (c *SolrMonitor) GetCollectionState(name string) (*CollectionState, error)
 }
 
 func (c *SolrMonitor) doGetCollectionState(name string) (*CollectionState, error) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
 	coll := c.collections[name]
+
 	if coll == nil {
 		return nil, nil
 	}
-	return coll.GetData()
+
+	return coll.collectionState, nil
+}
+
+func (c *SolrMonitor) GetQueryNodes() ([]string, error) {
+	if c.zkCli.State() != zk.StateHasSession {
+		return nil, errors.New("not currently connected to zk")
+	}
+
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	return append([]string{}, c.queryNodes...), nil
```

Comment:
Could we remove the accessor methods?

Reply:
Currently, solrman is referring that. With solrman refactoring we can remove that.

Reply:
SG, let's definitely iterate. I think these pieces can definitely be simplified with the event-driven model.

Reply:
Sure! we will do that